packages/settings-library/src/settings_library/postgres.py

```diff
@@ -57,6 +62,11 @@ def dsn(self) -> str:
         )
         return dsn
 
+    @cached_property
+    def dsn_with_async_sqlalchemy(self) -> str:
+        dsn = self.dsn.replace("postgresql", "postgresql+asyncpg")
```

Comment:
THOUGHT: Since we are going to deprecate aiopg this means that this will be the final ``dsn``. 
For that reason, I would directly implement it properly as

```python
    @cached_property
    def dsn_with_async_sqlalchemy(self) -> str:
        dsn: str = PostgresDsn.build(
            scheme="postgresql+asycnpg",
            user=self.POSTGRES_USER,
            password=self.POSTGRES_PASSWORD.get_secret_value(),
            host=self.POSTGRES_HOST,
            port=f"{self.POSTGRES_PORT}",
            path=f"/{self.POSTGRES_DB}",
        )
        return dsn
```
You can even add a deprecation warning if using ``dsn`` ...

The ``replace`` in addition is dangerous ... for instance, what if any of the attributes, i.e. user or password ,etc is ``postgresql``??